gravitee-io/gravitee-access-management
# fix: create a new authentication context when applying multiple idps

Closes: https://github.com/gravitee-io/issues/issues/6916

To test: 

You need to create 2 http identity providers:
- Configure 1 with MD5 and Hex and actual working authentication url `/authenticate?username={#principal}&password={#credentials}`
- The other with a non working url (even not existing)

Then mock the authentication service (make sure the MD5 + Hex password matches)
```json
{
  "mappings": [
    {
      "request": {
        "method": "GET",
        "url": "/authenticate?username=username&password=f925916e2754e5e03f75dd58a5733251",
        "queryParameters": {
          "username": {
             "equalTo": "username"
          },
          "password" : {
            "equalTo" : "f925916e2754e5e03f75dd58a5733251"
          }
        }
      },
      "response": {
        "status": 200,
        "jsonBody": {
          "sub" : "032f602b-87bf-4ef2-af60-2b87bfaef2f2",
          "updated_at" : "2022-01-05T16:43:21.109Z",
          "auth_time" : 1641494137,
          "preferred_username" : "username",
          "given_name" : "Name",
          "family_name" : "LastName",
          "email" : "usename@email.com"
        },
        "headers": {
          "Content-Type": "application/json"
        }
      }
    },
    {
      "request": {
        "method": "POST",
        "url": "/authenticate"
      },
      "response": {
        "status": 400,
        "jsonBody":
          {
          "message" : "bad request"
          },
        "headers": {
          "Content-Type": "application/json"
        }
      }
    }
  ]
}
```

Then massively call your gateway
```
curl -XPOST http://localhost:8092/domain/oauth/token\?grant_type\=password\&fakeVar\=\[\1-1000] --form 'username=username' --form 'password=Test@123' --form 'client_id=<client_id>'
```
